packages/next/src/lib/metadata/types/metadata-interface.ts

```diff
@@ -451,11 +451,13 @@ interface Metadata extends DeprecatedMetadataFields {
   classification?: null | string
 
   /**
-   * Arbitrary name/value pairs for the document.
+   * Arbitrary name/value pairs for the document, can be an object or an array of objects.
    */
-  other?: {
-    [name: string]: string | number | Array<string | number>
-  } & DeprecatedMetadataFields
+  other?:
+    | ({
+        [name: string]: string | number | Array<string | number>
+      } & DeprecatedMetadataFields)
+    | Array<{ name?: string; property?: string; content?: string | number }>
```

Comment:
Something like;
```suggestion
    | Array<{ name: string; content: (string | number)[] } | { property: string; content: (string | number)[] }>
```

Reply:
Ok, I have split `name` and `property` into a union type.

Reply:
I think the option of content possibly being an array (like the "old" one) should stick, bit of DX

Reply:
My idea is that the previous object-form API could not support multiple values with the same name, so it is necessary to support array-form content. 

If the `other` field supports array form, developers can pass in multiple elements with the same name but different values, which allows developers to achieve the required functionality while keeping the API more concise.

```ts
other: [
  { name: 'content', content: `value1` },
  { name: 'content', content: `value2` },
  { name: 'other content', content: 'value3' }
]
```

Reply:
```ts
other: [
  { name: 'content', content: [`value1`, `value2`] },
  { name: 'other content', content: 'value3' }
]
```